tests/PHPUnit/Framework/TestCase/SystemTestCase.php

```diff
@@ -726,10 +726,9 @@ protected static function restoreDbTables($tables)
         DbHelper::truncateAllTables();
 
         // insert data
-        $existingTables = DbHelper::getTablesInstalled();
         foreach ($tables as $table => $rows) {
             // create table if it's an archive table
-            if (strpos($table, 'archive_') !== false && !in_array($table, $existingTables)) {
+            if (strpos($table, 'archive_') !== false && !DbHelper::tableExists($table)) {
```

Comment:
might it be more performant to manually do a `SHOW TABLES like 'archive_'` to fetch all existing archive tables at once, instead of firing the query for each table?

Reply:
:+1: